lib/jnpr/junos/device.py

```diff
@@ -412,6 +422,99 @@ def open(self, *vargs, **kvargs):
             originating ``Exception`` is assigned as ``err._orig``
             and re-raised to the caller.
         """
+        #
+        #  PRIVATE Functions
+        #
+
+        def _get_connected_slot():
+            """return connected slot. In case of error return None"""
+
+            rsp = self.rpc.get_chassis_inventory()
+            if rsp.tag == 'error':
+                raise RuntimeError('Could not determine RE slot id')
+            if rsp.tag == 'output':
+                # this means that there was an error; due to the
+                # fact that this connection is not on the master
+                # @@@ need to validate on VC-member
+                self._reRole = "backup"
+            else:
+                self._reRole = "master"
+
+            rsp = self.rpc.get_route_engine_information()
+            if rsp.tag == 'error':
+                raise RuntimeError('Could not determine RE slot id')
+            re_list = rsp.xpath('.//route-engine')
+            if len(re_list) == 1:
+                return "RE0"
+            for re in re_list:
+                m_state = re.find('mastership-state')
+                if self._reRole.lower() == m_state.text.lower():
+                    x_slot = re.find('slot')
+                    if x_slot is None:
+                        raise RuntimeError("Could not find RE slot id")
+                    return "RE" + str(x_slot.text)
+            raise RuntimeError("Could not find RE slot id")
+
+        def _connect_to_other_re():
+
+            # Determine other re
+            other_re = None
+            if self._connected_re.upper() == 'RE0':
+                other_re = 're1'
+            else:
+                other_re = 're0'
+
+            # get the name of other re
+            command = etree.XML('<configuration><groups><name>%s</name>'
```

Comment:
This assumes the RE has a unique hostname configured, and the host executing the PyEZ script can resolve this unqualified hostname to a unique IP address, has connectivity to that IP address, etc. There are a LOT of assumptions here which I don't think are safe to make.